docs(animations): add type interfaces

sean-perkins opened the pull request:
- Removes the "Browser Support" section from the animation docs
- Rewords the summary
- Organizes the API related types, methods, etc. under an "API" section with a description of what it pertains to
  - Adds missing type information for the `AnimationBuilder`

liamdebeasi commented on docs/utilities/animations.md:
````diff
@@ -1560,14 +1560,258 @@ Due to a bug in Safari versions 9-11, stepping through animations via `progressS
+
+:::
+
+### Animation
+
+```tsx
+interface Animation {
+  parentAnimation?: Animation;
+  elements: HTMLElement[];
+  childAnimations: Animation[];
+  id: string | undefined;
+
+  /**
+   * Play the animation
+   *
+   * If the `sync` options is `true`, the animation will play synchronously. This
````
Isn't this the same information that is listed in the properties/methods tables below?

sean-perkins commented:
Reworked this again. I removed the duplicate `Animation` interface definition. Organized the subsections under "API" to help make it clearer that the `Animation` type for `AnimationBuilder` is referring to the animation/the current docs page.

liamdebeasi commented on docs/utilities/animations.md:
```diff
@@ -16,11 +16,11 @@ import TabItem from '@theme/TabItem';
+Ionic Animations is a tool that enables developers to create complex animations in a platform-agnostic manner, without requiring a specific framework or an Ionic app.
 
-Building efficient animations can be tricky. Developers are often limited by the libraries available to them as well as the hardware that their apps run on. On top of that, many animation libraries use a JavaScript-driven approach to running animations where they handle the calculation of your animation's values at every step in a `requestAnimationFrame` loop. This reduces the scalability of your ...
+Creating efficient animations can be challenging, as it is limited by the available libraries and hardware resources of the app. Moreover, many animation libraries use a JavaScript-driven approach, which can reduce the scalability of animations and use up CPU time.
```
A few small nits:

- Animations are limited by the hardware resources of the device the app runs on, not the app itself.
- What does "it" refer to in "as it is limited"? The original wording referred to developers, but "it" seems to imply the animation
